rclpy/src/test/rclpy_common/test_handle.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "rclpy_common/handle.h"
+
+void int_destructor(void * p)
+{
+  *reinterpret_cast<int *>(p) = 1;
```

Comment:
@ivanpauno nit: consider turning this into an increment, so that a destructor being called twice does not goes by unnoticed.